CMakeLists.txt

```diff
@@ -1,10 +1,141 @@
 if (WINDOWS)
-cmake_minimum_required(VERSION 3.17.0)
+cmake_minimum_required(VERSION 3.17)
 else()
-cmake_minimum_required(VERSION 3.9.4)
+cmake_minimum_required(VERSION 3.12)
 endif()
 
-project(OPENTIMELINEIO_ROOT)
+#------------------------------------------------------------------------------
+# Project Meta data
+# TODO: read this information from a configuration file, here, and in setup.py
 
-add_subdirectory(src)
+set(OTIO_VERSION_MAJOR "0")
+set(OTIO_VERSION_MINOR "13")
+set(OTIO_VERSION_PATCH "0")
+set(OTIO_VERSION       ${OTIO_VERSION_MAJOR}.${OTIO_VERSION_MINOR}.${OTIO_VERSION_PATCH})
 
+set(OTIO_AUTHOR "Contributors to the OpenTimelineIO project")
+set(OTIO_AUTHOR_EMAIL "opentimelineio@pixar.com")
+set(OTIO_LICENSE "Modified Apache 2.0 License")
+
+project(OpenTimelineIO VERSION ${OTIO_VERSION} LANGUAGES C CXX)
+
+#------------------------------------------------------------------------------
+# Options
+
```

Comment:
can you add a note that the idea with this section is that any `option()` calls in the cmake system be put here and only here